libs/remix-ui/editor/src/lib/remix-ui-editor.tsx

```diff
@@ -541,6 +542,31 @@ export const EditorUI = (props: EditorUIProps) => {
       }
     })
 
+    editor.onDidPaste((e) => {
+      if (e && e.range && e.range.startLineNumber >= 0 && e.range.endLineNumber >= 0 && e.range.endLineNumber - e.range.startLineNumber > 10) {
+        const modalContent: AlertModal = {
+          id: 'newCodePasted',
+          title: 'New pasted code detected!',
+          message: (
+            <div> <i className="fas fa-exclamation-triangle text-danger mr-1"></i>
+              Remix detected you have just pasted a consequent code snippet or contract in the editor.
+              <div>
+                Please make sure you fully understand this new code before executing <b>any</b> transaction that uses it. 
+                <div>
+                Your wallet might be exposed and at risk if you run untrusted code. In a worst-case scenario you could loose all your money.
```

Comment:
It may be good to highlight sensitive parts of the message e.g `risk`, `losing money`, etc.

Reply:
The language needs a slight update:

You have just pasted a code snippet or contract in the editor.
Make sure you fully understand this new code before executing any transaction that uses it.
Your wallet might be exposed and at risk if you run untrusted code. In a worst-case scenario, you could loose all your money.
If you don't fully understand it, please don't run this code.
If you are not a smart contract developer, please ask assistance from someone you trust and has the appropriate skills to determine whether this code is safe to use.